File: internal/models/log.go
package models

import (
	"time"
)

type Log struct {
	ID               uint      `gorm:"primaryKey" json:"id"`
	Latency          int       `gorm:"column:latency;not null" json:"latency"`
	PromptTokens     int       `gorm:"column:promptTokens;default:0;not null" json:"promptTokens"`
	CompletionTokens int       `gorm:"column:completionTokens;default:0;not null" json:"completionTokens"`
	CacheReadTokens  int       `gorm:"column:cacheReadTokens;default:0" json:"cacheReadTokens"`
	CacheWriteTokens int       `gorm:"column:cacheWriteTokens;default:0" json:"cacheWriteTokens"`
	TotalTokens      int       `gorm:"column:totalTokens;default:0;not null" json:"totalTokens"`
	Cost             int64     `gorm:"column:cost;default:0;not null" json:"cost"`
	Status           int       `gorm:"column:status;default:200;not null" json:"status"`
	ErrorMessage     string    `gorm:"column:errorMessage" json:"errorMessage"`
	APIKeyID         *uint     `gorm:"index;column:apiKeyId" json:"apiKeyId"` // Nullable for chat requests
	APIKey           *GatewayAPIKey `gorm:"foreignKey:APIKeyID;constraint:OnDelete:SET NULL" json:"apiKey,omitempty"`
	ModelName        string    `gorm:"column:modelName" json:"modelName"`
	ProviderName     string    `gorm:"column:providerName" json:"providerName"`
	OwnerChannelID   *uint     `gorm:"column:ownerChannelId" json:"ownerChannelId"`
	OwnerChannel     *Channel  `gorm:"foreignKey:OwnerChannelID" json:"ownerChannel,omitempty"`
	OwnerChannelUserID *uint   `gorm:"column:ownerChannelUserId" json:"ownerChannelUserId"`
	RequestPath      string    `gorm:"column:requestPath;type:varchar(255)" json:"requestPath"`
	RequestHeaders   string    `gorm:"column:requestHeaders;type:text" json:"requestHeaders"`  // JSON format
	ResponseHeaders  string    `gorm:"column:responseHeaders;type:text" json:"responseHeaders"` // JSON format
	CreatedAt        time.Time `gorm:"column:createdAt;index" json:"createdAt"`
}

func (Log) TableName() string {
	return "Log"
}

File: web/src/types/log.ts
// Log types
export interface APIKeyInfo {
  id: number
  name: string
  user?: {
    id: number
    email: string
    role: string
  }
}

export interface ChannelInfo {
  id: number
  name: string
  user?: {
    id: number
    email: string
  }
}

export interface Log {
  id: number
  model?: string
  modelName?: string
  model_name?: string
  provider?: string
  providerName?: string
  provider_name?: string
  latency_ms?: number
  latency?: number
  promptTokens?: number
  prompt_tokens?: number
  completionTokens?: number
  completion_tokens?: number
  totalTokens?: number
  total_tokens?: number
  tokens?: number
  cost: number
  status: number | string
  errorMessage?: string
  error_message?: string
  apiKeyId?: number
  apiKey?: APIKeyInfo
  ownerChannelId?: number
  ownerChannel?: ChannelInfo
  ownerChannelUserId?: number
  requestHeaders?: string
  responseHeaders?: string
  createdAt: string
  created_at?: string
}

export interface LogDetailData {
  requestBody?: string
  responseBody?: string
}

export interface LogDetailResponse {
  log: Log
  detail: LogDetailData | null
}

export interface LogDetail extends Log {
  detail?: LogDetailData
}

export interface LogListResponse {
  logs: Log[]
  total: number
  page: number
  limit: number
}